include/boost/beast/core/buffers_generator.hpp

```diff
@@ -0,0 +1,200 @@
+
+    @li An error occurs.
+
+    This operation is implemented in terms of one or more calls
+    to the stream's `write_some` function.
+
+    @param stream The stream to which the data is to be written.
+    The type must support the <em>SyncWriteStream</em> concept.
+
+    @param generator The generator to use.
+
+    @return The number of bytes written to the stream.
+
+    @throws system_error Thrown on failure.
+
+    @see BuffersGenerator
+*/
+template<
+    class SyncWriteStream,
+    class BuffersGenerator
+#if ! BOOST_BEAST_DOXYGEN
+    , typename std::enable_if<is_buffers_generator<
+        typename std::decay<BuffersGenerator>::
+            type>::value>::type* = nullptr
+#endif
+    >
+std::size_t
+write(
+    SyncWriteStream& stream,
+    BuffersGenerator&& generator);
+
+/** Write all output from a BuffersGenerator asynchronously to a
+    stream.
+
+    This function is used to write all of the buffers generated
+    by a caller-provided @ref BuffersGenerator to a stream. The
+    function call always returns immediately. The asynchronous
+    operation will continue until one of the following
+    conditions is true:
+
+    @li A call to the generator's `is_done` returns `false`.
+
+    @li An error occurs.
+
+    This operation is implemented in terms of zero or more calls
+    to the stream's `async_write_some` function, and is known as
+    a <em>composed operation</em>.  The program must ensure that
+    the stream performs no other writes until this operation
+    completes.
+
+    @param stream The stream to which the data is to be written.
+    The type must support the <em>SyncWriteStream</em> concept.
+
+    @param generator The generator to use.
+
+    @param handler The completion handler to invoke when the
+    operation completes. The implementation takes ownership of
+    the handler by performing a decay-copy. The equivalent
+    function signature of the handler must be:
+    @code
+    void handler(
+        error_code const& error,        // result of operation
+        std::size_t bytes_transferred   // the number of bytes written to the stream
+    );
+    @endcode
+    Regardless of whether the asynchronous operation completes
+    immediately or not, the handler will not be invoked from
+    within this function. Invocation of the handler will be
+    performed in a manner equivalent to using `net::post`.
+
+    @see BuffersGenerator
+*/
+template<
+    class AsyncWriteStream,
+    class BuffersGenerator,
+    class CompletionToken
+#if !BOOST_BEAST_DOXYGEN
+    , typename std::enable_if<is_buffers_generator<
+        BuffersGenerator>::value>::type* = nullptr
+#endif
+    >
+auto
+async_write(
+    AsyncWriteStream& stream,
+    BuffersGenerator generator,
+    CompletionToken&& token) ->
+        typename net::async_result<
+            typename std::decay<
+                CompletionToken>::type,
+            void(error_code, std::size_t)>::return_type;
+
+} // beast
```

Comment:
Of all the options it has, it seems I will not be able to teach clang-format to fix namespace comments without adding the `namespace` keyword.

Oh well, I'll just keep an eye out for this one. Like the other formatting glitches you fixed in the presence of preprocessor conditions

Reply:
You can leave the namespace word in, it doesn't hurt anything